packages/manager/modules/adp/src/deploy/deploy.controller.js

```diff
@@ -0,0 +1,644 @@
+
+    // Instances
+    const totalInstancesRequired = sumBy(nodes, node => get(node, 'count', 0));
+    const pciUsedInstances = get(quota, ['instance', 'usedInstances'], 0);
+    const pciTotalInstances = get(quota, ['instance', 'maxInstances'], 0);
+    const pciAvailableInstances = pciTotalInstances - pciUsedInstances;
+    const isInstancesValid = pciAvailableInstances - totalInstancesRequired >= 0;
+
+    // CPU
+    const totalCpuRequired = sumBy(nodes, node => get(node, ['selectedFlavor', 'vcpus'], 0));
+    const pciUsedCpu = get(quota, ['instance', 'usedCores'], 0);
+    const pciTotalCpu = get(quota, ['instance', 'maxCores'], 0);
+    const pciAvailableCpu = pciTotalCpu - pciUsedCpu;
+    const isCpuValid = pciAvailableCpu - totalCpuRequired >= 0;
+
+    const isClusterValid = isRamValid && isInstancesValid && isCpuValid;
+    if (!isClusterValid) {
+      const currentlyUsed = pciUsedRam / 1000;
+      const requiredForCluster = totalRamRequired / 1000;
+      const quotas = [
+        {
+          computeName: 'ram',
+          currentlyUsed,
+          requiredForCluster,
+          maxLimit: pciTotalRam / 1000,
+          required: currentlyUsed + requiredForCluster,
+          unit: 'gb',
+        },
+        {
+          computeName: 'cpu',
+          currentlyUsed: pciUsedCpu,
+          requiredForCluster: totalCpuRequired,
+          maxLimit: pciTotalCpu,
+          required: pciUsedCpu + totalCpuRequired,
+        },
+        {
+          computeName: 'nodes',
+          currentlyUsed: pciUsedInstances,
+          requiredForCluster: totalInstancesRequired,
+          maxLimit: pciTotalInstances,
+          required: pciUsedInstances + totalInstancesRequired,
+        },
+      ];
+      this.showQuotaInsufficientErrorModel(
+        this.selectedPublicCloud.description,
+        this.selectedPublicCloud.project_id,
+        quotas,
+      );
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * show error model with details about current configuration and available quota
+   *
+   * @param {*} publicCloudName
+   * @param {*} publicCloudId
+   * @param {*} quotas
+   */
+  showQuotaInsufficientErrorModel(publicCloudName, publicCloudId, quotas) {
+    this.cucControllerHelper.modal.showModal({
+      modalConfig: {
+        template: errorModel,
+        controller: errorModelController,
+        controllerAs: '$ctrl',
+        windowClass: 'modal-dialog-centered',
+        size: 'lg',
+        backdrop: true,
+        resolve: {
+          publicCloudName: () => publicCloudName,
+          publicCloudId: () => publicCloudId,
+          quotas: () => quotas,
+        },
+      },
+    });
+  }
+
+  /** #####################################################################
+      STORAGE
+   * ##################################################################### */
+
+  initStorage() {
+    this.storage.initialized = true;
+    this.storage.hdfs_effective_storage = this.nodesConfig.worker.raw_storage_min_gb;
```

Comment:
Imo `snakeCase` seems quite unusual for Javascript naming

Reply:
The API accepts parameters named like that in `snakeCase`. But I have changes the corresponding local variables to use `camelCase`